packages/kit/src/runtime/server/index.js

```diff
@@ -40,6 +41,31 @@ export async function respond(incoming, options, state = {}) {
 		locals: {}
 	};
 
+	if (options.methodOverride.enabled && request.method.toUpperCase() === 'POST') {
+		const { strategy = '', key: method_key = '', allowedMethods = [] } = options.methodOverride;
+		let new_request_method;
+
+		if (
+			['both', 'form_data'].includes(strategy) &&
+			request.body instanceof ReadOnlyFormData &&
+			request.body.has(method_key)
+		) {
+			new_request_method = (request.body.get(method_key) || request.method).toUpperCase();
+		}
+
+		if (['both', 'url_parameter'].includes(strategy) && incoming.query.has(method_key)) {
+			new_request_method = (incoming.query.get(method_key) || request.method).toUpperCase();
+		}
+
+		if (
+			new_request_method &&
+			allowedMethods.includes(new_request_method) &&
+			new_request_method !== 'GET'
```

Comment:
I think in addition to checking `GET` here it should also be validated in the `allowedMethods` validator. Because otherwise a user might want to add `GET` in `allowedMethods` and wonder why it is silently ignored. The validator would then have a descriptive error explaining that `GET` is not allowed ever.

But then it probably doesn't need to be explicitly checked here, when it will never be in the allowed list.

Maybe also throw `HEAD` in the mix? At this point I'm not sure if these cases are needed if `allowedMethods` exists.

Reply:
I've updated the validator to check for `GET` in `allowedMethods` and removed the check for `GET` in this file.